### PR TITLE
Logging - fix formatting

### PR DESCRIPTION
**Pull request by eliska-n**
continues on this branch: https://github.com/TeskaLabs/asab/pull/576

The biggest disadvantage is the `copy()` mechanism used. It does not work without it.

If merged, we could start sing colors in docker again :)

### CODE REVIEW

**Comment by eliska-n** [comment]
not yet fully tested, I'll come back to this one ASAP

**Review by ateska — APPROVED** [review]

**Comment by eliska-n** [comment]
@ateska I'm reviving this MR

There's a _loop_exception_handler method in the log.py that is unused. I'd like to delte it, but it has quite nice docstring looking important

**Comment by eliska-n** [comment]
don't know what is sd_id and what it is good for

**Comment by eliska-n** [comment]
I forgot what I did with the formatting in the structured data formatter

File: asab/log.py [hunk 1]
```diff
@@ -428,10 +427,29 @@ def __init__(self, fmt=None, datefmt=None, style='%', sd_id='sd'):
 		self.converter = time.gmtime
 
 
-class JSONFormatter(StructuredDataFormatter):
+class JSONFormatter(logging.Formatter):
+
+	INSTANCE_ID = os.environ.get("INSTANCE_ID")
```
Comment (eliska-n) [inline asab/log.py:432]: this is crucial for analysis of the data. it should be added to the log in other formatters, too. OR why not to have it in a logger (i.e. BEFORE it gets to formatting)

File: asab/log.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ async def schedule(app, interval):
 					elif format == '5micro':
 						self.SyslogHandler.setFormatter(SyslogRFC5424microFormatter(sd_id=Config["logging"]["sd_id"]))
 					elif format == 'json':
-						self.SyslogHandler.setFormatter(JSONFormatter(sd_id=Config["logging"]["sd_id"]))
+						self.SyslogHandler.setFormatter(JSONFormatter())
```
Comment (eliska-n) [inline asab/log.py:156]: json format is configured in [logging:syslog] config section. But it is not a syslog, isn't it? What is it?

**Comment by eliska-n** [comment]
For the formatting only - the aim is to have colorful logs in console and not colorful in logs sent... somewhere...
That should be possible now - because the formatters do not influence each other (because they don't modify the record).

So, I'll delete the "running_in_container" check and let's see what happens

File: asab/log.py [hunk 3]
```diff
@@ -428,10 +418,29 @@ def __init__(self, fmt=None, datefmt=None, style='%', sd_id='sd'):
 		self.converter = time.gmtime
 
 
-class JSONFormatter(StructuredDataFormatter):
+class JSONFormatter(logging.Formatter):
+
+	INSTANCE_ID = os.environ.get("INSTANCE_ID")
+	SERVICE_ID = os.environ.get("SERVICE_ID")
+	NODE_ID = os.environ.get("NODE_ID")
+	HOSTNAME = socket.gethostname()
```
Comment (ateska) [inline asab/log.py:426]: `INSTANCE_ID` etc. is not mandatory ..

**Review by ateska — CHANGES_REQUESTED** [review]

**Comment by eliska-n** [comment]
Tests needed:

- [ ] in development
- [ ] in docker container
- [ ] in our UI (Maintenance > Services >> terminal
- [ ] in syslog
- [ ] in json

File: asab/log.py (hunk 2 above)
Comment (eliska-n) [reply asab/log.py:156]: that's ok

File: asab/log.py (hunk 1 above)
Comment (eliska-n) [reply asab/log.py:423]: nope, we dont want to have the ids in console

**Pull request merged**